dev/bench/odbc/main.cc

```diff
@@ -0,0 +1,186 @@
+        std::cerr << "checkError function was called with an invalid handle!!"
+                  << std::endl;
+        return 1;
+      }
+
+      if ((retcode != SQL_SUCCESS) && (retcode != SQL_SUCCESS_WITH_INFO)) {
+        wprintf(L"ERROR: %d: %ls : %ls\n", nativeError, sqlState, errMsg);
+      }
+
+      errNum++;
+    }
+
+    wprintf(L"%ls\n", errmsg);
+    return 1;
+  }
+
+  return 0;
+}
+
+#define CHECK_OK(EXPR, ERROR)                                        \
+  do {                                                               \
+    auto ret = (EXPR);                                               \
+    if (checkError(ret, SQL_HANDLE_DBC, dbc, (SQLWCHAR*)L##ERROR)) { \
+      exit(1);                                                       \
+    }                                                                \
+  } while (0)
+
+int main(int argc, char** argv) {
+  SQLHENV env;
+  SQLHDBC dbc;
+  SQLHSTMT stmt;
+  SQLHSTMT stmt1;
+
+  std::string dsn(argv[0]);
+
+  SQLAllocHandle(SQL_HANDLE_ENV, SQL_NULL_HANDLE, &env);
+  // we want ODBC3 support, set env handle
+  SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION, (void*)SQL_OV_ODBC3_80, 0);
+  // allocate connection handle
+  SQLAllocHandle(SQL_HANDLE_DBC, env, &dbc);
+
+  // connect to the DSN using dbc handle
+  CHECK_OK(SQLDriverConnect(dbc, nullptr, (SQLCHAR*)(dsn.c_str()), SQL_NTS, nullptr, 0,
+                            nullptr, SQL_DRIVER_COMPLETE),
+           "Error -- Driver Connect failed");
+
+  std::ofstream timing_output("odbc_perf_record");
+
+  for (size_t iter = 0; iter < 100; iter++) {
+    CHECK_OK(SQLAllocHandle(SQL_HANDLE_STMT, dbc, &stmt),
+             "Error -- Statement handle alloc failed");
+
+    const auto start{std::chrono::steady_clock::now()};
+    CHECK_OK(SQLExecDirect(
+                 stmt,
+                 (SQLCHAR*)"SELECT * FROM SNOWFLAKE_SAMPLE_DATA.TPCH_SF1000.LINEITEM "
+                           "LIMIT 100000000",
+                 SQL_NTS),
+             "Error -- Statement execution failed");
+
+    const SQLULEN bulkSize = 10000;
+    CHECK_OK(SQLSetStmtAttr(stmt, SQL_ATTR_ROW_ARRAY_SIZE, (SQLPOINTER)bulkSize, 0),
+             "Error -- SetStmtAttr failed");
+
+    // bind columns to buffers
+    SQLINTEGER val_orderkey[bulkSize];
+    SQLINTEGER val_partkey[bulkSize];
+    SQLINTEGER val_suppkey[bulkSize];
+    SQLINTEGER val_linenumber[bulkSize];
+    SQLDOUBLE val_quantity[bulkSize];
+    SQLDOUBLE val_extendedprice[bulkSize];
+    SQLDOUBLE val_discount[bulkSize];
+    SQLDOUBLE val_tax[bulkSize];
+    SQLCHAR val_retflag[bulkSize][2];
+    SQLCHAR val_linestatus[bulkSize][2];
+    SQL_DATE_STRUCT val_shipdate[bulkSize];
+    SQL_DATE_STRUCT val_commitdate[bulkSize];
+    SQL_DATE_STRUCT val_receiptdate[bulkSize];
+    SQLCHAR val_shipinstruct[bulkSize][26];
+    SQLCHAR val_shipmode[bulkSize][11];
+    SQLCHAR val_comment[bulkSize][45];
+
+    CHECK_OK(SQLBindCol(stmt, 1, SQL_C_LONG, (SQLPOINTER)&val_orderkey,
```

Comment:
These calls to SQLBindCol should not take the address to each of the arrays. They should just use the arrays directly. 

You would need to take the address if you were binding scalar variables instead of arrays.